index.d.ts

```diff
@@ -680,6 +697,81 @@ declare global {
     FramePrototypingMixin,
     ReactionMixin {}
 
+  interface OpaqueNodeMixin  {
+    readonly id: string
+    readonly absoluteTransform: Transform
+    relativeTransform: Transform
+    x: number
+    y: number
+    readonly width: number
+    readonly height: number
+    readonly removed: boolean
+    remove(): void
+    readonly parent: (BaseNode & ChildrenMixin) | null
+    name: string
+    toString(): string
+
+    getPluginData(key: string): string
+    setPluginData(key: string, value: string): void
+
+    // Namespace is a string that must be at least 3 alphanumeric characters, and should
+    // be a name related to your plugin. Other plugins will be able to read this data.
+    getSharedPluginData(namespace: string, key: string): string
+    setSharedPluginData(namespace: string, key: string, value: string): void
+    setRelaunchData(data: { [command: string]: /* description */ string }): void
+  }
+
+  interface MinimalBlendMixin {
+    readonly opacity?: number
+    readonly blendMode?: BlendMode
+  }
+
+  interface TextSublayerNode {
```

Comment:
Could you try to deduplicate this with properties already in `TextNode`?

Reply:
Do you mean make TextNode extend this interface? Yes. Otherwise these properties are a subset of TextNode

Reply:
I find it easier to read as is. There's no clear patter as to what makes a property belong to a TextNode vs TextSublayerNode other than things in TextSublayerNode are what we decided (or are able) to support in FigJam-plugins/FigJam-UI

Reply:
It's a lot of duplication though. It's true that it's often hard to figure out exactly what properties are available on each node but 1) that's what the docs are for, a flattened representation 2) the typings file attempts to have the least amount of duplication, having a mix of approaches is the worst of both worlds

Reply:
Done. Yeah it was tough to go from jsvm-public-nodes to this while deduping this. Manual validation via scanning is hard when comparing to impl code base. But I agree with what you're saying.